src/_DataStructures_/LinkedList/index.js

```diff
@@ -125,24 +125,32 @@ class LinkedList {
     if (!this.head) {
       return null;
     }
+    if (index === 0) {
+      const node = this.head;
+      this.head = this.head.next;
+      this.size -= 1;
+      // set the next of the node null
+      node.next = null;
+      return node;
+    }
 
-    if (index >= this.length()) {
+    if (index >= this.size - 1) {
       return this.removeFromEnd();
     }
 
     let address = this.head;
     let previous = address;
     let count = index;
 
-    while (count) {
-      address = address.next;
+    while (count >= 1) {
       previous = address;
+      address = address.next;
       count -= 1;
     }
-
-    const node = address;
-    previous.next = address.next.next;
+    const node = previous.next;
```

Comment:
previous.next === address  after while loop . After loop `address` should be remove node and `previous` should be previous node of node to remove. Therefore i think it's better to have  `const node = address;` than `const node = previous.next;`

Reply:
Got it